OpenSim/Common/Component.cpp

```diff
@@ -838,26 +836,61 @@ void Component::constructOutputForStateVariable(const std::string& name)
             SimTK::Stage::Model);
 }
 
+// mark components owned as properties as subcomponents
+void Component::markPropertiesAsSubcomponents()
+{
+    // if being invoked either constructing a new Component
+    // or the properties have been modified. In the latter case
+    // we must make sure that pointers to old properties are cleared
+    _components.clear();
+
+    // Now mark properties that are Components as subcomponents
+    for (int i = 0; i < getNumProperties(); ++i) {
+        auto& prop = updPropertyByIndex(i);
+        if (prop.isObjectProperty()) {
+            for (int j = 0; j < prop.size(); ++j) {
+                Object& obj = prop.updValueAsObject(j);
+                if (Component* comp = dynamic_cast<Component*>(&obj) ) {
+                    markAsSubcomponent(comp);
+                }
+                //Set<Component>* objects = dynamic_cast<Set<Component>*>(&obj)
+                else {
+                    std::string objType = obj.getConcreteClassName();
+                    if (obj.hasProperty("objects")) {
+                        auto& objectsProp = obj.updPropertyByName("objects");
+                        for (int k = 0; k < objectsProp.size(); ++k) {
+                            Object& obj = objectsProp.updValueAsObject(k);
+                            if (Component* comp = dynamic_cast<Component*>(&obj) )
+                                markAsSubcomponent(comp);
```

Comment:
I currently have the marking as part of finalizeFromProperties(), which is already called recursively, but if we chose to decouple the two, then yes. I wanted to minimize what you have to call in the construct so that is my primary reason for lumping them.